### PR TITLE
Make S3 credentials session scoped

### PR DESCRIPTION
**Pull request by omo**
This is a follow up for #2247.

With this change, S3FS reads the credentials from ClientContext instead of Database.
Once this is in, we can change SET's default scope to SESSION from GLOBAL.

### CODE REVIEW

File: extension/httpfs/include/s3fs.hpp [hunk 1]
```diff
@@ -1,16 +1,15 @@
 #pragma once
 
 #include "httpfs.hpp"
-#include "duckdb/main/database.hpp"
+// TODO: Make this part of duckdb.h, in addition to other httpfs dependencies.
```
Comment (Mytherin) [inline extension/httpfs/include/s3fs.hpp:4]: What does this todo mean exactly?

**Review by Mytherin — APPROVED** [review]
Thanks for the PR! Looks great. I like the FileOpener solution to this problem. 

Just one minor question, otherwise I think it is ready to merge.

File: extension/httpfs/include/s3fs.hpp (hunk 1 above)
Comment (omo) [reply extension/httpfs/include/s3fs.hpp:4]: Oh, my point was that fle_opener.hpp has to be included in amalgamation-generated duckdb.h once s3fs becomes real extension. Maybe unnecessary comment. Let me remove it.

**Comment by omo** [comment]
I updated the branch to address the point. PTAL?
Thanks for the review Mark as always!

**Comment by omo** [comment]
OK, relevant checks have passed.

**Comment by omo** [comment]
Would you mind merging this when you have time? Thanks!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**